src/Microsoft.DotNet.Build.Tasks.Feed/BlobUrlInfo.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage.Auth;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks.Feed
+{
+    class BlobUrlInfo
+    {
+        private const string accountNameandEndpointRegex = @"(?<accountname>[a-z0-9]+)\.(?<endpoint>.+?)";
+        private const string containerAndBlobRegex = @"/(?<containername>[^\/]+)/(?<blobpath>.*)";
```

Comment:
These are missing the \^ anchor at the beginning to prevent the regex.  According to a tool that I just remember I wrote aeons ago, your container and blob regex has this behavior:

* /this/is/a/test (successful match)
  * containername = this
  * blobpath = is/a/test
* wait/this/is/a/test (successful match)
  * containername = this
  * blobpath = is/a/test

I'd expect that you don't actually want it to match the second case.

Similarly, your account name and endpoint regex is a successful match on each of these inputs:

* https://storagesample.blob.core.windows.net/mycontainer/blob1.txt
* storagesample.blob.core.windows.net
* .storagesample.blob.core.windows.net
* //storagesample.blob.core.windows.net

It matched all of them as `accountname`=`storagesample`, `endpoint`=`b`.  (Just "b")

So you also have a bug in that regex, since you later believe that you can use the pieces to reassemble the whole URI.

Reply:
@bartonjs The URI pieces (hostname and path) are matched against, not the full URL.  The URI pieces come out in a well formed fashion (no slashes) and the path segment is always prefixed with a slash.  Will add the anchor for completeness though.